docs/topics/wasm/wasm-js-interop.md

```diff
@@ -1,164 +1,227 @@
 
-let Counter = {
-    value: 0,
-    step: 1,
-    increment() {
-        this.value += this.step;
-    }
-};
+Consider this global JavaScript variable:
+
+```javascript
+let globalCounter = 0;
+```
+You can declare it in Kotlin using external `var` or `val` properties:
+
+```kotlin
+external var globalCounter: Int
+```
+These properties are initialized externally. The properties can't have `= value` initializers in Kotlin code.
 
+#### JavaScript classes
+
+Consider this JavaScript class:
+
+```javascript
 class Rectangle {
     constructor(height, width) {
         this.height = height;
         this.width = width;
     }
-
-    get area() {
-        return this.calcArea();
-    }
-
-    calcArea() {
+    area() {
         return this.height * this.width;
     }
 }
 ```
 
-Here's how you can use this JavaScript code in Kotlin:
+You can use it in Kotlin as an external class:
 
 ```kotlin
-// Kotlin/Wasm
+external class Rectangle(height: Double, width: Double) : JsAny {
+  val height: Double
+  val width: Double
+  fun area(): Double
+}
+```
+
+All declarations inside the `external` class are implicitly considered external.
 
-// Use external functions to call JS functions defined in global scope
-external fun consoleLogExample(): Unit
+#### External interfaces
 
-// In addition to functions, you can have external top-level properties
-external var externalInt: Int
+You can describe the shape of a JavaScript object in Kotlin. Consider this JavaScript function and what it returns:
 
-// External objects
-external object Counter {
-    fun increment(): Unit
-    val value: Int
-    var step: Int
+```javascript
+function createUser(name, age) {
+    return { name: name, age: age };
 }
+```
+See how its shape can be described in Kotlin with an `external interface User` type:
 
-// External class
-external class Rectangle(height: Double, width: Double) {
-    val height: Double
-    val width: Double
-    val area: Double
-    fun calcArea(): Double
+```kotlin
+external interface User : JsAny {
+  val name: String
+  val age: Int
 }
+
+
+external fun createUser(name: String, age: Int): User
 ```
 
-See the full code in the example
-project [Kotlin/Wasm browser](https://github.com/Kotlin/kotlin-wasm-examples/tree/main/browser-example).
+External interfaces don't have runtime type information and are a compile-time-only concept.
+Therefore, external interfaces have some restrictions compared to regular interfaces:
+* You can't use them on the right-hand side of `is` checks.
+* You can't use them in class literal expressions (such as `User::class`).
+* You can't pass them as reified type arguments.
+* `as` casts to external interfaces always succeed.
 
-> Some "external" Kotlin/JS features are not supported in Kotlin/Wasm:
-> * Implementing or extending external types
-> * External [enum classes](enum-classes.md)
->
-{type="note"}
 
-### @JsFun annotation
+#### External objects
+
+Consider this JavaScript's variables holding an object:
```

Comment:
Not sure that we need to use this apostrophe. We usually say "Kotlin variable", "Python variable" but not Kotlin's, Python's. 

It is also can be re-written as "variable in Javascript that holds and object".